### PR TITLE
[ncp] deliver ACK frame to host side.

### PR DESCRIPTION
**Pull request by bukepo**
This PR delivers ACK frames to host side by appending this frame content and its metadata to the reply of sending raw stream request.

### CODE REVIEW

File: src/core/api/link_raw_api.cpp [hunk 1]
```diff
@@ -444,6 +444,15 @@ void LinkRaw::InvokeTransmitDone(otRadioFrame *aFrame, bool aFramePending, otErr
 #endif
 }
 
+#if OPENTHREAD_CONFIG_LEGACY_TRANSMIT_DONE
+void LinkRaw::InvokeTransmitDone(const otRadioFrame *aFrame, bool aFramePending, otError aError)
+{
+    // A trick to indicate whether FramePending is set.
+    const otRadioFrame *ackFrame = aFramePending ? aFrame : NULL;
```
Comment (chshu) [inline src/core/api/link_raw_api.cpp]: Would it be better to generate a pseudo ACK frame? FYI: https://github.com/openthread/openthread/blob/master/examples/platforms/posix/radio.c#L632

Comment (bukepo) [inline src/core/api/link_raw_api.cpp]: Good idea, I will generate this ACK frame in those platforms not supporting passing ACK frame yet.

**Comment by bukepo** [comment]
This PR will be updated after #2020 merged.

File: src/ncp/ncp_base.cpp [hunk 2]
```diff
@@ -1161,27 +1162,59 @@ void NcpBase::LinkRawReceiveDone(otRadioFrame *aFrame, otError aError)
...
+            SuccessOrExit(OutboundFrameFeedData(aAckFrame->mPsdu, aAckFrame->mLength));
+            SuccessOrExit(OutboundFrameFeedPacked(
+                SPINEL_DATATYPE_INT8_S
+                SPINEL_DATATYPE_INT8_S
+                SPINEL_DATATYPE_UINT16_S
+                SPINEL_DATATYPE_STRUCT_S(   // PHY-data
+                    SPINEL_DATATYPE_UINT8_S // 802.15.4 channel
+                    SPINEL_DATATYPE_UINT8_S // 802.15.4 LQI
+                ),
+                aAckFrame->mPower,          // TX Power
```
Comment (jwhui) [inline src/ncp/ncp_base.cpp]: nit: this should be RSSI for the received ACK frame, right?

File: include/openthread/platform/radio.h [hunk 3]
```diff
@@ -406,7 +406,7 @@ extern void otPlatRadioTxStarted(otInstance *aInstance, otRadioFrame *aFrame);
 
 /**
  * The radio driver calls this method to notify OpenThread that the transmission has completed,
- * this callback pass up the ACK frame, new add platforms should use this callback function.
+ * this callback pass up the ACK frame, new platforms should use this callback function.
```
Comment (jwhui) [inline include/openthread/platform/radio.h]: Suggestion.

Before:
```
The radio driver calls this method to notify OpenThread that the transmission has completed,
this callback pass up the ACK frame, new platforms should use this callback function.
```

After: 
```
The radio driver calls this function to notify OpenThread that the transmit operation has completed,
providing both the transmitted frame and, if applicable, the received ack frame.
```

File: src/core/api/link_raw.hpp [hunk 4]
```diff
@@ -96,7 +96,7 @@ class LinkRaw
      * This method invokes the mTransmitDoneCallback, if set.
      *
      */
-    void InvokeTransmitDone(otRadioFrame *aFrame, bool aFramePending, otError aError);
+    void InvokeTransmitDone(const otRadioFrame *aFrame, const otRadioFrame *aAckFrame, otError aError);
```
Comment (jwhui) [inline src/core/api/link_raw.hpp]: Not necessarily in this PR, but we could improve the doxygen in this file (e.g., document parameters and add explicit `void`).

Comment (bukepo) [inline src/core/api/link_raw.hpp]: Thanks, may I finish this documentation task in a coming PR, since I've done part of this work in that PR.

**Pull request merged**